### PR TITLE
(BOLT-503) Support boltdir.

### PR DESCRIPTION
**Pull request by adreyer**
This adds support Boltdir a single directory to use as the default for
user supplied data to bolt. The configfile, inventoryfile and modules
now default to be inside the boltdir. This means the previous
configuration location ~/.puppetlab/bolt.yaml is deprecated in favor of
~/.puppetlabs/bolt/bolt.yaml. There is now a default for the modulepath
modules/ inside the boltdir.

### CODE REVIEW

File: lib/bolt/bolt_option_parser.rb [hunk 1]
```diff
@@ -0,0 +1,292 @@
+# frozen_string_literal: true
```
Comment (MikaelSmith) [inline lib/bolt/bolt_option_parser.rb]: Would be cool to have this refactor broken out as a separate commit.

File: lib/bolt/bolt_option_parser.rb [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+      define('--modulepath MODULES',
+             "List of directories containing modules, separated by '#{File::PATH_SEPARATOR}'") do |modulepath|
+        @options[:modulepath] = modulepath.split(File::PATH_SEPARATOR)
+      end
+      define('--boltdir FILEPATH',
+             'Specify what Boltdir to load config from (default: autodiscovered from current working dir)') do |path|
+        @options[:configfile] = path
+      end
+      define('--configfile FILEPATH',
+             'Specify where to load config from (default: ~/.puppetlabs/bolt.yaml)') do |path|
```
[the remark this answers is not in the document]
Comment (MikaelSmith) [reply lib/bolt/bolt_option_parser.rb]: Is this PR changing the default config location?

[the remark this answers is not in the document]
Comment (adreyer) [reply lib/bolt/bolt_option_parser.rb]: yes

**Comment by katelopresti** [comment]
Updated DITA source.

**Pull request merged**